agent.go

```diff
@@ -191,6 +200,35 @@ func (a *Agent) register() error {
 	return nil
 }
 
+// runMetrics is a long-running goroutine that exposes an http metrics interface
+func (a *Agent) runMetrics() {
+	if a.config.Telemetry.PrometheusRetentionTime < 1 || a.config.MetricsInterface != "" {
```

Comment:
It seems like if users configure `MetricsInterface` with a value, it will always return here (since it will not be an empty string). This doesn't match my understanding of the use of this config. Am I missing something?

Reply:
Yes, somehow I messed up this logic while trying to fix the tests...

Fixed.